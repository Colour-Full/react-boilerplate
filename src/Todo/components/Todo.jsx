// @flow
import * as React from 'react'
import type { Element } from 'react'
import Item from '@atlaskit/item'

export type TodoProps = {
  onClick: () => any,
  completed: boolean,
  text: string
}

const Todo = ({onClick, completed, text}: TodoProps): Element<any> => (
  <Item
    isCompact
    onClick={onClick}
    style={{
      textDecoration: completed ? 'line-through' : 'none'
    }}
  >
    {text}
  </Item>
)

export default Todo
