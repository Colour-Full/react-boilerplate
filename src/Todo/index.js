// @flow
import React, { type Element } from 'react'

// Import Components
import Footer from './components/Footer'

// Import Containers
import AddTodo from './containers/AddTodo'
import VisibleTodoList from './containers/VisibleTodoList'


const Todo = (): Element<any> => (
  <div>
    <AddTodo />
    <VisibleTodoList />
    <Footer />
  </div>
)

export default Todo
