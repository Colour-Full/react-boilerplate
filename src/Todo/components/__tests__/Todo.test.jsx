import React from 'react'
import Todo from '../Todo'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Item from '@atlaskit/item'

Enzyme.configure({ adapter: new Adapter() })

describe('A Todo Component ', () => {
  it('It should render without throwing an error ', () => {
    const props = {
      onClick: jest.fn(),
      completed: false,
      text: 'To do'
    }
    const wrapper = shallow(<Todo {...props} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should call onClick handler when Item component is clicked', () => {
    const props = {
      onClick: jest.fn(),
      completed: false,
      text: 'To do'
    }
    const wrapper = shallow(<Todo {...props} />)
    wrapper.find(Item).simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })
})
