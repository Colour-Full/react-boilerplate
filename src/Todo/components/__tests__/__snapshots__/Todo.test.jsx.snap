// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Todo Component  It should render without throwing an error  1`] = `
<Item
  autoFocus={false}
  description=""
  isCompact={true}
  isDisabled={false}
  isHidden={false}
  onClick={[MockFunction]}
  role="button"
  shouldAllowMultiline={false}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  To do
</Item>
`;
